tests/integration/test___init__.py

```diff
@@ -0,0 +1,273 @@
+    assert nested_dir_table_line in index_topic
+
+    # 9. docs with a documentation file added in the nested directory
+    caplog.clear()
+    nested_dir_doc_table_key = "nested-dir-doc"
+    (nested_dir_doc_file := nested_dir / "doc.md").write_text(
+        nested_dir_doc_content := "nested dir doc content 1"
+    )
+
+    urls_with_actions = run(
+        base_path=tmp_path, discourse=discourse_api, dry_run=False, delete_pages=True
+    )
+
+    assert len(urls_with_actions) == 3
+    (_, nested_dir_doc_url, _) = urls_with_actions.keys()
+    assert (urls := tuple(urls_with_actions)) == (doc_url, nested_dir_doc_url, index_url)
+    assert_substrings_in_string(chain(urls, ("'create'", "'success'")), caplog.text)
+    index_topic = discourse_api.retrieve_topic(url=index_url)
+    nested_dir_doc_table_line = (
+        f"| 2 | {nested_dir_doc_table_key} |"
+        f" [{nested_dir_doc_content}]({urlparse(nested_dir_doc_url).path}) |"
+    )
+    assert nested_dir_doc_table_line in index_topic
+    nested_dir_doc_topic = discourse_api.retrieve_topic(url=nested_dir_doc_url)
+    assert nested_dir_doc_topic == nested_dir_doc_content
+
+    # 10. docs with the documentation file in the nested directory removed in dry run mode
+    caplog.clear()
+    nested_dir_doc_file.unlink()
+
+    urls_with_actions = run(
+        base_path=tmp_path, discourse=discourse_api, dry_run=True, delete_pages=True
+    )
+
+    assert (urls := tuple(urls_with_actions)) == (doc_url, nested_dir_doc_url, index_url)
+    assert_substrings_in_string(chain(urls, ("'delete'", "'update'", "'skip'")), caplog.text)
+    index_topic = discourse_api.retrieve_topic(url=index_url)
+    assert nested_dir_doc_table_line in index_topic
+    nested_dir_doc_topic = discourse_api.retrieve_topic(url=nested_dir_doc_url)
+    assert nested_dir_doc_topic == nested_dir_doc_content
+
+    # 11. docs with the documentation file in the nested directory removed with page deletion
+    #     disabled
+    caplog.clear()
+
+    urls_with_actions = run(
+        base_path=tmp_path, discourse=discourse_api, dry_run=False, delete_pages=False
+    )
+
+    assert (urls := tuple(urls_with_actions)) == (doc_url, nested_dir_doc_url, index_url)
+    assert_substrings_in_string(chain(urls, ("'delete'", "'update'", "'skip'")), caplog.text)
+    index_topic = discourse_api.retrieve_topic(url=index_url)
+    assert nested_dir_doc_table_line not in index_topic
+    nested_dir_doc_topic = discourse_api.retrieve_topic(url=nested_dir_doc_url)
+    assert nested_dir_doc_topic == nested_dir_doc_content
+
+    # 12. with the nested directory removed
+    caplog.clear()
+    nested_dir.rmdir()
+
+    urls_with_actions = run(
+        base_path=tmp_path, discourse=discourse_api, dry_run=False, delete_pages=True
+    )
+
+    assert (urls := tuple(urls_with_actions)) == (doc_url, index_url)
+    assert_substrings_in_string(chain(urls, ("'delete'", "'update'", "'success'")), caplog.text)
```

Comment:
Should we also assert that some reference of the old nested_dir_doc is present in the logs ?

Reply:
Thats a good point, that check should be added shortly in addition to similar checks for the other cases

Reply:
Fixed